lib/refinement/analyzer.rb

```diff
@@ -50,10 +50,31 @@ def annotate_targets!
     #   Defaults to `false`
     # @param log_changes [Boolean] whether modifications to the scheme are logged.
     #   Defaults to `false`
+    # @param filter_scheme_for_build_action [:building, :testing]
+    #   The xcodebuild action the scheme is being filtered for. The currently supported values are
+    #   `:building` and `:testing`, with the only difference being `BuildActionEntry` are not
+    #   filtered out when building for testing, since test action macro expansion could
+    #   depend on a build entry being present.
+    #   Defaults to `:building`.
     # @return [Xcodeproj::XCScheme] a scheme whose unchanged targets have been removed
-    def filtered_scheme(scheme_path:, change_level: :full_transitive, filter_when_scheme_has_changed: false, log_changes: false)
+    def filtered_scheme(scheme_path:, change_level: :full_transitive, filter_when_scheme_has_changed: false, log_changes: false,
+                        filter_scheme_for_build_action: :building)
```

Comment:
should this default change to `testing` ?

Reply:
I'm ok with switching this to `:testing`, though I think I'd just leave it with no default and require it to be specified

Reply:
Then I'll have to add a required CLI flag, which seems... not ideal